### ISSUE TITLE
make VOCDetection dataset always use year 2012 with trainval image set

### ISSUE THREAD

**Issue by NickleDave** [NickleDave/visual-search-nets#74]
currently for `VOCDetection` the default `image_set` is `train`

https://github.com/NickleDave/visual-search-nets/blob/13372dcb5ff675d7d4fbdc84bc85cb2104c51310/src/searchnets/datasets/voc.py#L74

this has the effect of halving the available training data, since there are 5k images in `train` and a total of ~10k in `trainval`.

I.e., we use the `train` split created by `searchnets` but then filter it by which images from that split are in the specified `image_set`, which will be `image_set='train'`, only 5k images
https://github.com/NickleDave/visual-search-nets/blob/13372dcb5ff675d7d4fbdc84bc85cb2104c51310/src/searchnets/datasets/voc.py#L143
(`file_names` are from the `image_set` text file)

This also limits the size of our `val` and `test` sets (which are taken from the total `trainval` split, since the original paper uses all of training and validation set from VOC 2012)

So:
- [ ] remove arguments for `year` and `image_set` 
- [ ] instead set them as constants of `2012` and `trainval` respectively

**Comment by NickleDave** [comment]
actually I see now that `searchnets.train` creates the dataset using the `trainval` split
https://github.com/NickleDave/visual-search-nets/blob/13372dcb5ff675d7d4fbdc84bc85cb2104c51310/src/searchnets/train.py#L169

so we were using ~5k images in the training set but that's because the split is the same as in the VSD paper, where they use 50% of `trainval` for training their models, 25% for validation, and 25% for test

Definitely don't like the way I did this, but I see no reason to think it introduced errors or impaired training now that I realized I was using the whole dataset

Closing as "wontfix"
